stored_requests/multifetcher.go

```diff
@@ -0,0 +1,50 @@
+package stored_requests
+
+import (
+	"context"
+	"encoding/json"
+)
+
+// MultiFetcher is a Fetcher composed of multiple sub-Fetchers that are all polled for results.
+type MultiFetcher []Fetcher
+
+// FetchRequests implements the Fetcher interface for MultiFetcher
+func (mf *MultiFetcher) FetchRequests(ctx context.Context, ids []string) (map[string]json.RawMessage, []error) {
+	var errs []error
+	result := make(map[string]json.RawMessage, len(ids))
+	missingIDs := 0 // The number of missing IDs that each fetcher reported.
+	// If the number of errors == number of missing IDs, then it is likely that the errors are simply due
+	// to the IDs missing from the return result.
+	numIDs := len(ids)
+	// suspect fetchers ... fetchers that returned errors that don't match the number of missing IDs.
+	suspect := 0
+
+	// Loop over the fetchers
+	for _, f := range *mf {
```

Comment:
Depending on how many fetchers could be in one `MultiFetcher` and how much latency is involved, we may want to kick off this loop body as a goroutine so they can run concurrently and use a [`WaitGroup`](https://golang.org/pkg/sync/#WaitGroup) to wait until they're all done.

Reply:
The use case I have in my mind is moving from one backend to another, where the stored requests can't easily be copied live from one backend to the other. This allows a transition period where it will work as long as it exists in one of the locations. So that would be a max of 2. But good idea if we want to crank up the number of fetchers higher.